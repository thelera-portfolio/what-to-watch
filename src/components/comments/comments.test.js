import Comments from "./comments.jsx";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";

const comments = [
  {
    comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 7,
    rating: 11,
    user: {
      id: 11.7,
      name: `Pete`,
    }
  },
  {
    comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 74,
    rating: 1,
    user: {
      id: 2,
      name: `Ann`,
    }
  },
  {
    comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 0,
    rating: 131,
    user: {
      id: 7,
      name: `George`,
    }
  },
];

const user = {
  id: 2, name: `Kate`
};

it(`Comments is rendered correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <Comments
          comments={comments}
          date={`Sat Jul 18 2020 11:37:10`}
          id={8}
          rating={10}
          user={user}
        />
      </BrowserRouter>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
