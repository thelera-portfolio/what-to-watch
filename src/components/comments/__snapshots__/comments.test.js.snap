// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments is rendered correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Pete
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 8th 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        11
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Ann
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 8th 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        1
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            George
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 8th 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        131
      </div>
    </div>
  </div>
</div>
`;
